scabha/schema_utils.py

```diff
@@ -187,7 +193,10 @@ def clickify_parameters(schemas: Union[str, Dict[str, Any]],
             if io is outputs and not (schema.is_file_type and not schema.implicit):
                 continue
 
-            policies = OmegaConf.merge(default_policies, schema.policies)
+            # None should not take precedence in the merge
```

Comment:
This seems to break something in breifast. I get
```
### running stack_time_cube
# Traceback (most recent call last):
#   File "<string>", line 7, in <module>
#   File "/home/bester/software/breifast/breifast-tron/breifast/cubes.py", line 29, in <module>
#     from breifast.main import cli, schemas
#   File "/home/bester/software/breifast/breifast-tron/breifast/main.py", line 26, in <module>
#     from . import breifast, catalogs, cubes, image_tools
#   File "/home/bester/software/breifast/breifast-tron/breifast/image_tools.py", line 23, in <module>
#     def measure_source_spectrum(images: List\[str],
#   File "/home/bester/software/breifast/breifast-tron/breifast/main.py", line 23, in <lambda>
#     clickify_parameters(schemas.cabs.get(f'breifast.{schema_name}'))(func)
#   File "/home/bester/software/stimela2/scabha/schema_utils.py", line 197, in clickify_parameters
#     merge_policies = {k: v for k, v in schema.policies.items() if v is not None}
# AttributeError: 'ParameterPolicies' object has no attribute 'items'
2024-11-07 16:09:41 STIMELA.tron.stack-cds ERROR: step 'tron.stack-cds' has failed, aborting the recipe
```

Reply:
I can get around this by casting the ParameterPolicies object to a dict using dataclasses.asdict but again not sure if this will have detrimental consequences elsewhere

Reply:
Yep, that works, but I would rather you use ``if isinstance(schema.policies, ParameterPolicies)`` do the asdict thing else do the other thing rather than catching exceptions. But that;s because I generally don't like using exceptions for managing "normal" code flow. Exceptions should be reserved for true errors as much as possible -- this also makes debugging easier (when you enable breakpoints on exception raised).

Reply:
Yeah, this was a bit naughty. Not sure why I did it like that